### ISSUE TITLE
[RFC] Option to redirect after item create/edit success to show page

### ISSUE THREAD

**Issue by wesley6j** [marmelab/admin-on-rest#688]
If a resource has a show page, it may be more reasonable to redirect to show page after successful form submission.
Can we have:

an optional `afterSuccess` prop for `Create` and `Show`?
`<Create {...props} afterSuccess="show">`

or redirect to `Show` if it is available and redirect to `List` otherwise.

I can make a PR if either of them helps.

**Comment by kimkha** [comment]
I think you should make a PR, and link it to here. And then, some will review your code.

**Comment by scriptology** [comment]
It would be cool if it would be possible to redirect to another entity

**Comment by kimkha** [comment]
@fzaninotto What do you think? Or any suggestions?

**Comment by wesley6j** [comment]
Now redirect can be automatic, it looks for the first available pages in a list.
And it can be configured manually, it accepts `list`, `show`, `edit` and custom url.
#692

**Comment by kodepareek** [comment]
Additional possible use case for this feature. 

I have created a custom Menu to route 2 different list views List1 and List2 (with different filters) to the same resource. 
There are Edit buttons on both list views. 
On entering Edit from List2 and Saving, Edit is redirecting the page to the List1 while the expectation is to redirect to List2.

After going through the source code, I see that in the Saga for Update redirects to the basePath from Edit 
Edit calculates basePath 

```
    getBasePath() {
        const { location } = this.props;
        return location.pathname.split('/').slice(0, -1).join('/');
    }

```
I will probably have to write a custom List view at this point. Can the above enhancement also accomodate my use case please.

**Comment by fzaninotto** [comment]
Yes, see the discussion going on in #692

**Comment by wesley6j** [comment]
Some progress with an example:

- create a custom toolbar
```jsx
import Toolbar from 'admin-on-rest/mui/form/Toolbar';
import { SaveButton } from 'admin-on-rest';
const PostCreateToolbar = props => <Toolbar {...props} >
    <SaveButton label="Save and show" redirect="show" />
    <SaveButton />
</Toolbar>;
```

- add toolbar to post create

```jsx
export const PostCreate = ({ ...props }) => (
    <Create {...props}>
        <SimpleForm toolbar={<PostCreateToolbar />} >
...
```

**result**
![image](https://cloud.githubusercontent.com/assets/4378997/26758275/df7cc854-48af-11e7-8f77-fdefdd77e4b0.png)

Each button redirects to different page.

**Comment by fzaninotto** [comment]
Fixed by #692
